lib/lumberjack/beats/server.rb

```diff
@@ -375,17 +380,29 @@ def close
     end
 
     def data(map, &block)
-      block.call(map) if block_given?
+      block.call(map, identity_stream(map)) if block_given?
     end
 
     def reset_next_ack(window_size)
-      klass = (@version == Parser::PROTOCOL_VERSION_1) ? AckingProtocolV1 : AckingProtocolV2
+      klass = version_1? ? AckingProtocolV1 : AckingProtocolV2
       @ack_handler = klass.new(window_size)
     end
 
     def send_ack(sequence)
       @fd.syswrite(@ack_handler.ack_frame(sequence))
     end
+
+    def version_1?
+      @version == Parser::PROTOCOL_VERSION_1
+    end
+
+    def identity_stream(map)
+      [peer,
+       map.fetch("beats", {})["name"],
```

Comment:
Well, Actually it can be the default or the user configured value, it's just more 'data' to identity a stream.  The source and the first part of the stream with `peer` should be pretty unique since it contains the IP + connected port. 

is that too much/enough?

WDYT?